### PR TITLE
Enabled persistence for Firebase database & some other fixes

### PR DESCRIPTION
**Pull request by riggaroo**
Hi there, 
Thanks so much for this fun example of Android Things! I've made some improvements to the companion app, please see the pull request if you would like to merge the changes. 

Changes:
- Enabled persistence for Firebase database to speed up the loading of
the data on the client side (initial loading seems slow still but this will speed up further requests), I've also moved the initialization of the adapter into onCreate and onDestroy as it was reloading the data every time the app paused for a bit, this should also improve the performance a bit. 
- Updated version numbers of firebase dependencies from 9.4.0 to 10.0.1
- Made the companion app target minSdkVersion 21 as there was nothing stopping this from working on lower device levels.
- Code cleanup of doorbell_entry.xml to use better naming for Views.

Let me know if there is anything wrong with the contribution.

### CODE REVIEW

File: companionApp/src/main/res/layout/doorbell_entry.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <ImageView
         android:layout_width="220dp"
         android:layout_height="match_parent"
-        android:id="@+id/imageView1" />
+        android:id="@+id/imageViewCapture" />
```
Comment (proppy) [inline companionApp/src/main/res/layout/doorbell_entry.xml]: shouldn't `DoorbellEntryAdapter.java` also be updated with those new names?

**Review by proppy — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution!

I'll let @devunwired or @jdkoren comment on the Android tweaks.

File: companionApp/src/main/res/layout/doorbell_entry.xml (hunk 1 above)
Comment (riggaroo) [inline companionApp/src/main/res/layout/doorbell_entry.xml]: I have updated the pull request.

**Review by proppy — APPROVED** [review]

**Comment by proppy** [comment]
also pinging @shawngit  :)

File: companionApp/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,16 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.example.androidthings.doorbell">
+          package="com.example.androidthings.doorbell">
```
Comment (jdkoren) [inline companionApp/src/main/AndroidManifest.xml]: nit: extra whitespace

**Review by jdkoren — COMMENTED** [review]
UI changes seem fine, but I don't know anything about using Firebase and related adapters, so I'll defer to others who do.

File: companionApp/src/main/AndroidManifest.xml (hunk 2 above)
Comment (riggaroo) [inline companionApp/src/main/AndroidManifest.xml]: I have fixed this.

**Pull request closed without merging**